### ISSUE TITLE
File Uploads set to Delete files does not delete files when an incomplete submission is deleted

### ISSUE THREAD

**Issue by romainpoirier** [verbb/formie#634]
This happens on an incomplete submission, where a custom path (example: `forms/{formName|camel}/{userId}/{id}`) is set on the `Upload Location` settings of the `File Upload` field.

**Steps to reproduce**
1. Set `Settings` > `File Uploads` to `Delete files` (`/admin/formie/forms/edit/{id}#tab-settings`)
2. Setup a multi-pages forms with a File Upload field
2. Setup a custom path on the Upload Location settings (exemple: `forms/{formName|camel}/{userId}/{id}`)
3. Fill in a new submission
4. Delete this incomplete submission
5. Check assets

### Form settings

- Multi-page form: Yes
- Submission Method: Page Reload
- Client-side Validation: Yes
- Custom Form Templates: Yes

**Additional info**
- Plugin version: 1.4.23
- Craft version: 3.7.18.2
- Multi-site: Yes

**Comment by engram-design** [comment]
Fixed for the next release. To get this fix early, change your `verbb/formie` requirement in `composer.json` to:

```
"require": {
  "verbb/formie": "dev-craft-3 as 1.4.23",
  "...": "..."
}
```

Then run `composer update`.

**Comment by engram-design** [comment]
Please note that even if you have a custom path, Formie won't delete an entire folder of assets, only the ones its associated with the submission. I mention this due to our testing in https://github.com/verbb/formie/issues/627 has probably resulted in a good amount of assets in the same folder. The fix in there should remove ones you overwrite, but it won't automatically delete a folder as defined in your custom path - that's just too risky.

So if you're testing this, just make sure to start with an empty folder from scratch.

In addition, despite submissions supporting soft-delete, uploads are deleted as soon as a submission is soft-deleted. This isn't ideal by any means, as it means if you want to restore a soft-deleted submission, your files are gone permanently.

This is because there's no event when something is permanently deleted for us to remove the file for the file system. Refer to https://github.com/craftcms/cms/issues/5074

**Comment by romainpoirier** [comment]
`Please note that even if you have a custom path, Formie won't delete an entire folder of assets, only the ones its associated with the submission.`
Ok, thank you for pointing this out! In my case, I am using a custom path (`forms/{formName|camel}/{userId}/{id}`) to create a folder per submission, based on the submission id. After testing again, you're right, the files are correctly deleted after deleting the submission, and the folder is kept. Do you have any idea of how to purge these empties assets folders?

**Comment by engram-design** [comment]
Hmm, I don't I'm afraid. It's far too presumptuous of Formie to delete an empty folder, as people could be directing uploads anywhere - even to the root folder of a volume, which deleting that would be nasty business.

Unfortunately, you might have to live with an empty folder.

**Comment by engram-design** [comment]
Fixed in [1.4.24](https://github.com/verbb/formie/releases/tag/1.4.24)
